src/Chapter1.hs

```diff
@@ -554,7 +555,14 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z 
+    | x > y && x < z = x
+    | x < y && x > z = x
+    | y > x && y < z = y
+    | y < x && y > z = y
+    | z > x && z < y = z
+    | z < x && z > y = z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂

In your case, you can replace the last two cases with `otherwise = z` as it is the only possibility when pattern matching didn't fit on the previous cases 🙂 

```suggestion
    | otherwise = z
```